pkg/kn/plugin/manager.go

```diff
@@ -401,6 +401,10 @@ func findMostSpecificPluginInPath(dir string, parts []string, lookupInPath bool)
 		var nameParts []string
 		var commandParts []string
 		for _, p := range parts[0:i] {
+			// skip arguments with slashes
+			if strings.Contains(p, "/") || strings.Contains(p, "\\") {
```

Comment:
I don't think that you just skip it bit completely stop the loop when the first part appears that can not be used as a plugin part, otherwise your could call plugins accidentally, e.g.

```
kn myplugin not/possible bla
```

would also call an existing plugin `kn-myplugin-bla`, but this is not following the plugin contract.

Also, you should ensure that you only filter characters that are not possible on the current's OS filesystem. E.g. on UNIX you can have a backslash as part of a file name, so a plugin like `kn-myplugin-ext\ra` is possible on UNIX (but not on Windows). So you should filter on the concrete file separator.

Reply:
> Also, you should ensure that you only filter characters that are not possible on the current's OS filesystem. E.g. on UNIX you can have a backslash as part of a file name, so a plugin like `kn-myplugin-ext\ra` is possible on UNIX (but not on Windows). So you should filter on the concrete file separator.

+1 that's a great point.